lib/cdo/data/logging/rake_task_event_logger.rb

```diff
@@ -62,11 +76,44 @@ def log_event(event, duration = nil, exception = nil)
     rescue => e
       Honeybadger.notify(
         e,
-        error_message: "Failed to log rake task information",
+        error_message: "Failed to log rake task information in firehose",
+        context: {
+          event: event
+        }
+      )
+    end
+  end
+
+  def log_cloud_watch(event, duration_ms, exception = nil)
+    unless @enable_cloudwatch
+      return
+    end
+    begin
+      metrics = {
+        metric_name: event,
+        value: duration_ms.nil? ? 1 : duration_ms,
+        dimensions: {name: "rake_performance",
+                     environment: rack_env,
+                     commit_hash: RakeUtils.git_revision,
+                     task_name: @rake_task.name,
+                     depth: @@depth,
+                     is_drone_run: ENV['CI'] ? true : false}
```

Comment:
Do we already consistently use the `environment` and `is_drone_run` elsewhere, or is this a new concept?

I think we'd like to head towards something more like an environment type and environment name. The value of `rack_env` is more of an environment type, and the actual environment would be something like prod, test, staging, levelbuilder, i18n, adhoc-<branch>, drone-<commit> etc.

Is this an opportunity to start using that terminology?

Reply:
hmmm not sure if I follow entirely what you mean here.

Maybe I shouldn't use the word "drone"

but as far as I know, the rack_env is widely used and the ENV['CI'] is how he have made the distinction between a drone run vs a non drone run for testing. 

I don't think we have an environment type? @sureshc  do you know about an example here?

Reply:
The Rails term `environment` (such as `RAILS_ENV`) really means "What TYPE of environment is this?" (staging, test, production) and not "What is the unique identifiers for this environment. We have multiple environments that all declare the same environment type. For example, `test-studio.code.org` uses `RAILS_ENV=test` but so does our local development environment when a developer executes unit tests locally, and so does our Continuous Integration environment (which currently uses Drone, but likely will transition to something else).

We've been talking about making a clear distinction between a type of environment and the name of a specific environment. To help support that, perhaps this Dimension should be named `environment_type` if its value is being set from the variable `rack_env`. Alternately, we could set the value of the `environment` Dimension to a unique name. This could be the AWS Stack Name for a "managed" environment, or its fully qualified domain name (`staging-studio.code.org`). Continuous Integration (Drone) builds might need the Drone Build ID in the `environment` to ensure uniqueness, as a single commit could be built multiple times in our Continuous Integration systems.

We already have legacy variables named `CIRCLE_CI_[somethjing]` even though we transitioned from CircleCI to Drone a few years ago. Perhaps we should name the Dimension `is_drone_run` to `is_continuous_integration_run`.

Reply:
We now have the subdomain and base domain names as Parameters in the CloudFormation Stack. Together the fully qualified domain name is a unique identifier. And within a single AWS Account, the subdomain, alone, is effectively a unique identifier.